dig.go

```diff
@@ -113,6 +115,9 @@ func (c *Container) provideInstance(val interface{}) error {
 	if vtype == _errType {
 		return errors.New("can't provide errors")
 	}
+	if vtype.Implements(_parameterObjectType) {
+		return errors.New("can't provide parameter objects")
```

Comment:
I've been using the term "parameter object" for types that embed `dig.Param` in all the documentation and errors so far. That's the standard name for objects which represent all the parameter of a function, created solely to avoid having too many arguments on the function. See http://wiki.c2.com/?ParameterObject, https://refactoring.com/catalog/introduceParameterObject.html.

We can call this something else, maybe give it a dig-specific name, but "parameterized object" has a whole different meaning.